crates/nu-command/tests/commands/select.rs

```diff
@@ -279,3 +279,90 @@ fn select_single_row_with_variable() {
     assert_eq!(actual.out, "[[a]; [3]]".to_string());
     assert!(actual.err.is_empty());
 }
+
+#[cfg(test)]
+mod range {
+    use nu_test_support::nu;
+
+    #[test]
+    fn positive_positive() {
+        let actual = nu!("[[a b c]; [1 2 3] [4 5 6] [7 8 9]] | select 1..2 | to nuon");
+
+        assert_eq!(actual.out, "[[a, b, c]; [4, 5, 6], [7, 8, 9]]");
+    }
+
+    // Should these two tests work? if negative vals would work this would translate to rows 1 0 4 3
+    #[test]
+    #[ignore]
+    fn positive_negative() {
+        let actual = nu!(
+            "[[a b c]; [1 2 3] [4 5 6] [7 8 9] [10 11 12] [13 14 15]] | select 1..(-2) | to nuon"
+        );
+
+        assert_eq!(
+            actual.out,
+            "[[a, b, c]; [4, 5, 6], [7, 8, 9], [10, 11, 12], [13, 14, 15]]"
+        );
+    }
+
+    // TODO `select` could maybe take values from the end of the list.
+    #[test]
+    #[ignore]
+    fn negative_negative() {
+        let actual = nu!("[[a b c]; [1 2 3] [4 5 6] [7 8 9]] | select (-1)..(-2) | to nuon");
+
+        assert!(actual.err.is_empty());
+    }
+
+    // Should these two tests work?
+    #[test]
+    #[ignore]
+    fn negative_positive() {
+        let actual = nu!("[[a b c]; [1 2 3] [4 5 6] [7 8 9]] | select (-1)..2 | to nuon");
+
+        assert_eq!(actual.out, "[[a, b, c]; [4, 5, 6], [7, 8, 9]]");
+    }
+
+    #[test]
+    #[ignore]
+    fn infinity_positive() {
+        let actual = nu!("[[a b c]; [1 2 3] [4 5 6] [7 8 9]] | select 1.. | to nuon");
+
+        assert_eq!(actual.out, "[[a, b, c]; [4, 5, 6], [7, 8, 9]]");
+    }
+
+    #[test]
+    #[ignore]
+    fn infinity_negative() {
+        let actual = nu!("[[a b c]; [1 2 3] [4 5 6] [7 8 9]] | select (-2).. | to nuon");
+
+        assert_eq!(actual.out, "[[a, b, c]; [4, 5, 6], [7, 8, 9]]");
+    }
+
+    #[test]
+    #[ignore]
+    fn ninfinity_positive() {
+        let actual = nu!("[[a b c]; [1 2 3] [4 5 6] [7 8 9]] | select ..2 | to nuon");
+
+        assert_eq!(actual.out, "[[a, b, c]; [4, 5, 6], [7, 8, 9]]");
```

Comment:
not sure this is correct, i would expect something like
```nushell
actual = [[a b c]; [1 2 3] [4 5 6] [7 8 9]] | select ..1
```
and
```nushell
expected = [[a b c]; [1 2 3] [4 5 6]]
```
because
```nushell
> ..1
─┬─
0│0
1│1
─┴─
```

Reply:
ok, let's figure this one out before doing the other ones, i think i might be confused because i'm thinking about all this in terms of Python slices, not Nushell ranges :cry:

Reply:
for instance, i expected `..(-5)` to select everything up until the *-5th* element of the input data, but that's incorrect, right? :confused:

Reply:
if i'm just not understanding Nushell ranges at all, we can close this :wink:

Reply:
I've added some general tests. 
the Current the implementation works only partially
( meaning only finite ranges work)
and select currently doesn't support negative ranges so that also needs to be implemented.

I'm having some problems, since the `run` function parses the data, and `select` then operates on them.
The problem occurs when i need to pass the data to `select` but need it to get the size of the list, column, data getting passed in. `PipelineData` doesn't implement any `clone` or `copy`  so i cannot copy the values, and to get length i need to change `PipelineData` into an iterator, which consumes it.

I'm stating to think i bit a little more than i can chew :sweat_smile: since I've stumbled into a little bit of a dead end.

Reply:
@amtoine You can learn more about ranges just by trying the ranges that I listed above with the `ls` and `range` command like `ls | range 0..3`. Then just substitute the other ranges and see what they do.

Reply:
Some of the test implementations are not final, and have mistakes, like this one :sweat_smile: 
I will try to implement it over the weekend :grin: